# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineBailouts.cpp
# Commit: 12d4b2481567
# Full Hash: 12d4b24815672b4b54ceb54e422a7d7d1d64542e
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2022-03-26 21:33:56
# Description:
#   Bug 1761492 - Override ResumeMode for exception bailouts. r=iain
#   
#   When bailing out to catch an exception, the ResumeMode from the snapshot doesn't
#   make sense as we're resuming at the start of a catch-block.
#   
# ==============================================================================

diff -r e2b40064eb59 -r 12d4b2481567 js/src/jit/BaselineBailouts.cpp
--- a/js/src/jit/BaselineBailouts.cpp	Sat Mar 26 09:53:47 2022 +0000
+++ b/js/src/jit/BaselineBailouts.cpp	Sat Mar 26 10:25:15 2022 +0000
@@ -545,12 +545,16 @@
   }
   prevFramePtr_ = virtualPointerAtStackOffset(0);
 
-  // Get the pc. If we are handling an exception, resume at the pc of the
-  // catch or finally block.
-  pc_ = catchingException() ? excInfo_->resumePC()
-                            : script_->offsetToPC(iter_.pcOffset());
+  // Get the pc and ResumeMode. If we are handling an exception, resume at the
+  // pc of the catch or finally block.
+  if (catchingException()) {
+    pc_ = excInfo_->resumePC();
+    resumeMode_ = mozilla::Some(ResumeMode::ResumeAt);
+  } else {
+    pc_ = script_->offsetToPC(iter_.pcOffset());
+    resumeMode_ = mozilla::Some(iter_.resumeMode());
+  }
   op_ = JSOp(*pc_);
-  resumeMode_ = mozilla::Some(iter_.resumeMode());
 
   return true;
 }
